Downsampling separated from download and carried

somsubhojit1988 opened the pull request:
Down sampling carried out after download.
Down sampling overwrites clip file.
TestDownSample - for testing newly added DownSample class.

Commit:
Downsampling separated from download and carried
out after download.
Downsampling overwrites clip file.
TestDownSample - for testing newly added DownSample class.

lukefahr commented:
What if I call:
```
asb = AudioSetBuilder( sampling_rate = 2000)
asb.getClips(100)

asb2 = AudioSetBuilder( sampling_rate = 8000)
asb2.getClips(100)
```

my reading is that with asb2, I will get 100 clips with sampling_rate = 2000 rather than 8000.  The reason is the files are overwritten on disk at 2000, so there is no way to recover the 8000.  

One solution may be check their sampling rate, and if it is too low to automatically re-download them.

somsubhojit1988 commented:
Right I will take care of that
Do you think it makes more sense if the original downloaded clips are preserved (with whatever their frame rates are) and then each run of the application creates separate set of the files renamed (pre/post append the corresponding frame-rates that are requested.
- So in that sort of a design when u download 100 clips setting frame-rate to 2000.
The data directory contains 200 files (f1, f2, .. f100 & f1_2khz .. f100_2khz). 
- Next run the download phase can be skipped as we already have then and a new set of 100 files aka: (f1_8khz .. f100_8khz) are created.

lukefahr commented:
I think there are three correct ways of doing it: 

1) You download the file, downsample, store.  When loading the files, check the sampling rate.  If the requested sampling rate is the same, great.  If it's higher than the existing file, you throw out the existing file, redownload it, and re-downsample to the new rate.  If it's lower than the existing file, you just downsample, store the new downsampled file, and return it.  

2)  You store the full sampling rate file on disk and downsample on the fly.  This seems the easiest to impliment, but might have the highest overhead.  I'm not sure if this will cause huge overheads or not. 

3) Your proposed method of storing **both** the origional and downsampled signals.  I might put the downsampled files in a separate sub-folder for tidyness.  

-A

Commit:
Downsampled files placed in respective directories.
Downsampled files are placed in directories named after sampling rate.
Downloaded files are unmodified.
Skipping downsampling beyond range 1000-41000
